dev/Interop/StoragePickers/PickerCommon.cpp

```
@@ -249,4 +253,62 @@ namespace PickerCommon {
     }
+
+    /// <summary>
+    /// Configure the FileSaveDialog, this is only for FileSavePicker.
+    /// </summary>
+    /// <param name="dialog"></param>
+    void PickerParameters::ConfigureFileSaveDialog(winrt::com_ptr<IFileSaveDialog> dialog)
+    {
+        winrt::hstring fileNameToSet;
+        if (!IsHStringNullOrEmpty(SuggestedFileName))
+        {
+            fileNameToSet = SuggestedFileName;
+        }
+        
+        if (!PickerCommon::IsHStringNullOrEmpty(SuggestedSaveFilePath))
+        {
+            auto suggestedPath = std::filesystem::path{ SuggestedSaveFilePath.c_str() };
+            if (!suggestedPath.empty())
+            {
+                // Force the directory of SuggestedSaveFilePath (if exist)
+                //      to be the starting location of save picker Dialog
+                auto dirPath = suggestedPath.parent_path();
+                if (!dirPath.empty())
+                {
+                    winrt::com_ptr<IShellItem> shellItem;
+                    HRESULT hr = SHCreateItemFromParsingName(dirPath.c_str(), nullptr, IID_PPV_ARGS(shellItem.put()));
+                    if (SUCCEEDED(hr))
+                    {
+                        check_hresult(dialog->SetFolder(shellItem.get()));
+                    }
+                    else
+                    {
+                        // The messagebox won't block user from using the FileSavePicker Dialog.
+                        // TODO: the message and label here needs globalization.
+                        MessageBoxW(
+                            this->HWnd,
+                            std::format(L"{} is unavailable. If the location is on this PC, make sure the device or drive is connected or the disc is inserted, and then try again. If the location is on a network, make sure you\x2019re connected to the network or Internet, and then try again. If the location still can\x2019t be found, it might have been moved or deleted.",
```

1. this is not localizable
2. why are we using messagebox inside the app sdk?
3. which activation context is this going to use to ensure this uses latest comctl?
4. more importantly, why show an error at all? either fail the call, fail silently (pick another folder), or create the folder

---

Offline discussion decided not to throw an error box for `suggested` value.